scalyr_agent/monitor_utils/k8s.py

```diff
@@ -2183,15 +2200,22 @@ def query_api(self, path):
             # We suppress warnings here to avoid spam about an unverified connection going to stderr.
             # This method of warning suppression is not thread safe and has a small chance of suppressing warnings from
             # other threads if they are emitted while this request is going.
-            with warnings.catch_warnings():
-                warnings.simplefilter("ignore", category=Warning)
-                response = self._session.get(url, timeout=self._timeout, verify=False)
-            if self._kubelet_url.startswith("https://"):
-                global_log.warn(
-                    "Accessing Kubelet with an unverified HTTPS request.",
-                    limit_once_per_x_secs=3600,
-                    limit_key="unverified-kubelet-request",
-                )
+            verify_connection = self._verify_connection()
+            if not verify_connection:
+                with warnings.catch_warnings():
```

Comment:
We could actually make our own context manager which uses ``warnings.catch_warnings()`` underneath. This would save us a couple of lines of code (and come handy if we want to use the same pattern in multiple places).

Then we could do something like:

```python
with context_managers.ignore_insecure_warnings(verify_connection=verify_connection):
   ....
```

Not a blocker for now, but something we should consider if we use this pattern in multiple places. And it would make "unit level" testing a bit easier as well.